h2o-algos/src/test/java/hex/pca/PCATest.java

```diff
@@ -442,4 +453,59 @@ public void testPCAPredMojoPojoNumericsOnly() {
       Scope.exit();
     }
   }
+
+  @Test
+  public void testPCAPipeline() throws IOException {
+    try {
+      Scope.enter();
+      Frame train = parse_test_file(Key.make("arrests.hex"), "smalldata/pca_test/USArrests.csv");
+      Scope.track(train);
+
+      pcaParameters._train = train._key;
+      pcaParameters._k = 4;
+      pcaParameters._transform = DataInfo.TransformType.NONE;
+      pcaParameters._pca_method = PCAParameters.Method.GramSVD;
+
+      PCAModel pca = new PCA(pcaParameters).trainModel().get();
+      Scope.track_generic(pca);
+
+      Frame reduced = pca.score(train);
+      Scope.track(reduced);
+
+      KMeansModel.KMeansParameters kmeansParameters = new KMeansModel.KMeansParameters();
+      kmeansParameters._train = reduced._key;
+      kmeansParameters._k = 2;
+
+      KMeansModel kmeans = new KMeans(kmeansParameters).trainModel().get();
+      Scope.track_generic(kmeans);
+
+      Frame clusters = kmeans.score(reduced);
+      Scope.track(clusters);
+
+      URI pcaMojoUri = pca.exportMojo(tmp.newFolder("pca").getAbsolutePath() + "/pca.zip", false);
+      URI kmeansMojoUri = kmeans.exportMojo(tmp.newFolder("kmeans").getAbsolutePath() + "/kmeans.zip", false);
+      File pipelineFile = tmp.newFile("pipeline.zip");
+
+      new MojoPipelineBuilder()
+              .addModel("pca", new File(pcaMojoUri))
+              .addMapping("PC1", "pca", 0)
+              .addMapping("PC2", "pca", 1)
+              .addMapping("PC3", "pca", 2)
+              .addMapping("PC4", "pca", 3)
+              .addMainModel("kmeans", new File(kmeansMojoUri))
+              .buildPipeline(pipelineFile);
+
+      GenericModel pipelineModel = Generic.importMojoModel(pipelineFile.getAbsolutePath(), true);
```

Comment:
Test the dummy metrics are in place ?

Reply:
We do not actually calculate any metrics. The metric builder is only needed for BigScore to work.